saleor/plugins/webhook/shipping.py

```diff
@@ -0,0 +1,161 @@
+    for webhook in webhooks:
+        response_data = trigger_webhook_sync(
+            event_type,
+            payload,
+            webhook.app,
+            EXCLUDED_SHIPPING_REQUEST_TIMEOUT,
+        )
+        # TODO handle a case when we didn't receive a proper response
+        if response_data:
+            excluded_methods.extend(
+                get_excluded_shipping_methods_from_response(response_data)
+            )
+    cache.set(cache_key, (payload, excluded_methods), CACHE_EXCLUDED_SHIPPING_TIME)
+    return parse_excluded_shipping_methods(excluded_methods)
+
+
+def get_excluded_shipping_data(
+    event_type: str,
+    previous_value: List[ExcludedShippingMethod],
+    payload_fun: Callable[[], str],
+    cache_key: str,
+) -> List[ExcludedShippingMethod]:
+    """Exclude not allowed shipping methods by sync webhook.
+
+    Fetch excluded shipping methods from sync webhooks and return them as a list of
+    excluded shipping methods.
+    The function uses a cache_key to reduce the number of
+    requests which we call to the external APIs. In case when we have the same payload
+    in a cache as we're going to send now, we will skip an additional request and use
+    the response fetched from cache.
+    The function will fetch the payload only in the case that we have any defined
+    webhook.
+    """
+
+    excluded_methods_map: Dict[str, List[ExcludedShippingMethod]] = defaultdict(list)
+    webhooks = _get_webhooks_for_event(event_type)
+    if webhooks:
+        payload = payload_fun()
+
+        excluded_methods_map = get_excluded_shipping_methods_or_fetch(
+            webhooks, event_type, payload, cache_key
+        )
+
+    # Gather responses for previous plugins
+    for method in previous_value:
+        excluded_methods_map[method.id].append(method)
+
+    # Return a list of excluded methods, unique by id
+    excluded_methods = []
+    for method_id, methods in excluded_methods_map.items():
+        reason = None
+        if reasons := [m.reason for m in methods if m.reason]:
+            reason = " ".join(reasons)
+        excluded_methods.append(ExcludedShippingMethod(id=method_id, reason=reason))
+    return excluded_methods
+
+
+def get_excluded_shipping_methods_from_response(
+    response_data: dict,
+) -> List[dict]:
+    excluded_methods = []
+    for method_data in response_data.get("excluded_methods", []):
+        try:
+            raw_id = method_data["id"]
+            typename, _id = graphene.Node.from_global_id(raw_id)
+            if typename == "app":
+                method_id = raw_id
+            elif typename == "ShippingMethod":
+                method_id = _id
+            else:
+                raise ValueError(
+                    f"Invalid type received. Expected ShippingMethod, got {typename}"
+                )
+        except (KeyError, ValueError, TypeError) as e:
+            logger.warning(f"Malformed ShippingMethod id was provided: {e}")
+            continue
```

Comment:
Lol we are raising Value error to catch it 3 lines later to call `continue` :D

Reply:
That felt like the cleanest way to keep a single logger statement. Suggestions are welcome ;)

Reply:
Why not just:
```
else:
    log.warning(f"Invalid type received. Expected ShippingMethod, got {typename}")
    continue
```